src/main/java/com/coreos/jetcd/cluster/MemberAddResponse.java

```diff
@@ -34,7 +30,17 @@ public Member getMember() {
    * returns a list of all members after adding the new member.
    */
   public List<Member> getMembers() {
+    List<Member> members = membersRef.get();
+    if (members == null) {
+      synchronized (membersRef) {
+        members = membersRef.get();
+        if (members == null) {
+          members = Util.toMembers(getResponse().getMembersList());
+          membersRef.lazySet(members);
+        }
+      }
+    }
+
```

Comment:
Does not give any effective advantage as the method is synchronized and it introduces an AtomicReference object which is allocated even not needed.